### ISSUE TITLE
Java: Unexpected Function Invocation Behavior For Java SDK 

### ISSUE THREAD

**Issue by zzzhy** [microsoft/semantic-kernel#5854]
I test the function invocation on the version of 1.0.0-rc2 by a simple plugin.

TimePlugin: just one method which supply a string of current time
when I call the getTime() function, it returns `2020-10-10 19:53:05`, or maybe some other time in the past years, which is not my expected result.

I need some help to fingure this out, thanks.
the code look like the followings:

```java
public String getTime() {
    KernelPlugin getCurrentTime = KernelPluginFactory.createFromObject(new TimePlugin(), "time");
    // add plugin into the kernel
    Kernel textCompletionKernel = textCompletionKernelBuilder.withPlugin(getCurrentTime).build();
    KernelFunction<String> kf = KernelFunctionFromPrompt.<String>builder()
         .withTemplate("tell me the current time in the format: yyyy-MM-dd HH:mm:ss")
         .withName("getCurrentTime")
         .withOutputVariable(new OutputVariable("the current time string", "java.lang.String"))
         .build();
    return textCompletionKernel.invokeAsync(kf)
         // tell to use the getCurrentTime function
        .withToolCallBehavior(ToolCallBehavior.requireKernelFunction(getCurrentTime.get("getCurrentTime")))
        .withResultType(ContextVariableTypes.getGlobalVariableTypeForClass(String.class))
        .mapNotNull(FunctionResult::getResult)
        .block();
}

public static class TimePlugin implements Serializable {
    @DefineKernelFunction(name = "getCurrentTime",
            description = "获取当前时间",
            returnType = "java.lang.String",
            returnDescription = "返回当前时间，格式为yyyy-MM-dd HH:mm:ss",
            samples = {@SKSample(inputs = "", output = "2024-04-01 01:01:01")})
    public String getCurrentTime() {
        LocalDateTime now = LocalDateTime.now();
        return DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss").format(now);
    }
}
```

**Comment by johnoliver** [comment]
This is a bug with required tool call invocation, effectively it gets requested multiple times, a fix is in #5962 and your sample works for me with that patch

**Comment by zzzhy** [comment]
@johnoliver Thank you for the latest update. May I ask when the next 1.x Maven jar will be released（Will it still be RC version）?

**Comment by johnoliver** [comment]
@zzzhy we are already at 1.0.1, so we are out of RC, I would guess a release next week as we have fixed a few things since 1.0.1

**Comment by bx-h** [comment]
I believe the issue lies with the `textCompletionKernelBuilder`. When I switched to using `chatCompletionKernel`, everything worked fine. Upon reviewing the `OpenAIChatCompletion.java` file, I found the following code, which is responsible for handling tool calls. (The `internalChatMessageContentsAsync` function)
<img width="673" alt="image" src="https://github.com/microsoft/semantic-kernel/assets/22412942/09b86a24-487f-4b21-832e-1b6ca11147f0">

However, in `OpenAITextGenerationService.java`, there is no code related to tool processing.
<img width="771" alt="image" src="https://github.com/microsoft/semantic-kernel/assets/22412942/c3ae76f1-422d-4471-9c95-404ebf7c50dc">
